### DISCUSSION TITLE
Documentation: Approaching `1.0` final updates for code and documentation (ideas and wishlist)

### DISCUSSION THREAD

**Discussion by bms63** [pharmaverse/admiral#2085] [General]
Hi @pharmaverse/admiral 

Hoping this thread will become a running list of ideas, wishes, desires, problems  around updates for Vignettes, Documentation and Readme.  

**BACKSTORY:**  I put in an issue for updating the `generic.Rmd` vignette for `derive_extreme_events()`.  However, as Stefan pointed out we should wait till we are more stable to make this update.  It got me thinking that we should start a running list of other updates to vignettes, readmes, documentation, etc that we might want to discuss and track.

I like this process of using Discussions to really nail down the work that should be done and then creating an issue that links to the discussion.  That way we don't have such vague and difficult issues to tackle.

**Comment by bms63** [comment]
# Update Generic.Rmd

Do we want to add anything to the https://pharmaverse.github.io/admiral/cran-release/articles/generic.html guide on `derive_extreme_events()`?
> 
> Any thoughts on this one @bundfussr ?

Sorry, I missed that one. In principle, yes. However, I would expect that we have to update the vignette significantly once the cleanup is complete. Thus I wouldn't do it now.

_Originally posted by @bundfussr in https://github.com/pharmaverse/admiral/issues/2015#issuecomment-1700974810_

**Reply by bms63** [reply]
`join` arguments need some additional discussion in vignette

**Comment by bms63** [comment]
# Clarify Release Schedule on ReadMe.

**Comment by manciniedoardo** [comment]
Noting down here a common idea from various of our calls: we should refresh our programming strategy so that it's up-to-date ahead of 1.0.

**Reply by bms63** [reply]
We split this out into another discussion - going to hide

**Comment by bms63** [comment]
# Records versus Observations

https://github.com/pharmaverse/admiral/discussions/2066#discussioncomment-7070534

**Comment by bms63** [comment]
# 1.0 means 100% test coverage

Once all the code updates are done for 1.0 I think it is worthwhile to bump our test coverage up to 100%

**Comment by bms63** [comment]
# Rehab the Reference Page 

Discussion around the Reference Page - We have a lot of functions!  Some are not user-facing but we exposed them to allow us to question and track them if they are needed.  However, with 1.0 I  believe we can start hiding them again.

Hoping we can come up with the list here to suppress and/or consolidate and then make into an issue. 

@pharmaverse/admiral

**Reply by bms63** [reply]
# The Others Section could be reduced

![image](https://github.com/pharmaverse/admiral/assets/10111024/19e4f663-00c8-4d34-9e19-a923e3d6b0a6)

**Reply by bms63** [reply]
# Remove the Utilities for Quosures Section

![image](https://github.com/pharmaverse/admiral/assets/10111024/c28ef281-6f59-4421-b4b2-92460ad89833)

**Reply by bms63** [reply]
# Remove the re-export section 

Personally, I think this is a bit of an advanced concept that a user doesn't need to know about

![image](https://github.com/pharmaverse/admiral/assets/10111024/41f53bee-5033-4ae8-adc5-92f95ad7b7dc)

**Reply by bms63** [reply]
Remove Deprecated Section and replace with Superseded Section

Since we are trying to no longer deprecate functions, only have argument deprecation, I think this section is not useful anymore.  We also do a great job in our changelog documenting deprecation.  However, I think the superseded section should replace the deprecation section with an explanation.

![image](https://github.com/pharmaverse/admiral/assets/10111024/c8e0b227-7aff-4e8e-aa94-c93691c8f906)

**Reply by bundfussr** [reply]
I think we can change most of the functions in the "Other" section to internal functions and stop exporting them.

However, I would list all exported function on the "Reference" page. If an exported function is not on the "Reference" page, it could confuse users and health authorities. I would wonder if it is safe to use a function which is not on the "Reference" page.

**Reply by bundfussr** [reply]
I would restructure the "Reference" page such that it has a similar structure as in the [admiral Function Overview](https://phuseaccount.sharepoint.com/:w:/r/sites/Pharmaverse-admiral/Shared%20Documents/Development%20Team%20Working%20Docs/admiral%20Functions%20Overview.docx?d=w64c31636c93c47dda066585be7a8c42b&csf=1&web=1&e=vKZTbb). Then the functions with the biggest impact are at the beginning.

**Reply by bms63** [reply]
Decision: We are going to no longer export developer-focused functions, e.g. `assert_db_requirements()`.  Use `internal` for kewords.  Continue to have a high-level of documentation even if not exported.

* [assert_db_requirements()](https://pharmaverse.github.io/admiral/dev/reference/assert_db_requirements.html) - Remove
* [assert_terms()](https://pharmaverse.github.io/admiral/dev/reference/assert_terms.html) - Remove
* [assert_valid_queries()](https://pharmaverse.github.io/admiral/dev/reference/assert_valid_queries.html) - Remove
* [extend_source_datasets()](https://pharmaverse.github.io/admiral/dev/reference/extend_source_datasets.html) - Remove
* [filter_date_sources()](https://pharmaverse.github.io/admiral/dev/reference/filter_date_sources.html) - Remove
* [format(<basket_select>)](https://pharmaverse.github.io/admiral/dev/reference/format.basket_select.html) - Export, but change keyword to internal
* [validate_basket_select()](https://pharmaverse.github.io/admiral/dev/reference/validate_basket_select.html) - Remove
* [validate_query()](https://pharmaverse.github.io/admiral/dev/reference/validate_query.html) - Remove
* [chr2vars()](https://pharmaverse.github.io/admiral/dev/reference/chr2vars.html) - Remove
* [signal_duplicate_records()](https://pharmaverse.github.io/admiral/dev/reference/signal_duplicate_records.html) - Export, but change keywords to internal
* [extract_duplicate_records()](https://pharmaverse.github.io/admiral/dev/reference/extract_duplicate_records.html) - Export, but change keywords to internal


Issue created for this one: https://github.com/pharmaverse/admiral/issues/2220#issue-1979841756

**Reply by bms63** [reply]
# Condense Utilities into only 1 section

**Reply by bms63** [reply]
# Review Example Datasets

![image](https://github.com/pharmaverse/admiral/assets/10111024/4b247208-b91e-4d4d-9d32-4a9c86d02ed3)

**Comment by bms63** [comment]
# Partial Matching for arguments

I believe we have had a few arguments sneak through that are actually not arguments in our functions - either in examples or templates. - just due to typos.  I believe R has some partial matching for argument to help deal with this, but we might want to turn this off.  There is a way to set this I have been told, but unclear on it.  @pharmaverse/admiral any memories?

**Reply by bundfussr** [reply]
As far as I know, you can't turn it off. But you can set the `warnPartialMatchArgs` option. Then a warning is issued if partial matching for arguments is used,

**Reply by bms63** [reply]
where does one set this option?

**Reply by bundfussr** [reply]
`options(warnPartialMatchArgs = TRUE)`

**Reply by bms63** [reply]
place into `.Rprofle`?

**Comment by bundfussr** [comment]
# Improve "See also"

I would propose to change the usage of the `@family` tag. At the moment it replicates the section of the "Reference" page. Therefore you can't leave the section by the links provided in the "See also" section.

I would use more than one `@family` tag. For instance, for `derive_summary_records()` we could use `@family derv_obs` and `@family derv_sum`. The first one lists all derivations adding observations and the second one lists all derivations which summarise like `derive_summary_records()` and `derive_var_merged_summary()`.

**Reply by bms63** [reply]
Ummm...I vaguely remember talking about this two years ago with the multiple family tags.  I also think there was a reason we had to pin the family and keywords together.  Happy to try another way to display information.

**Reply by bundfussr** [reply]
@bms63 , as far as I remember you suggested to pin the family and keywords tag together but I do not remember the reason.

**Reply by bms63** [reply]
lol - well I have no memory of why I chose this path.  happy to experiment if it improves documentation.   maybe I just chose to make it easier as an initial roll out???

**Reply by manciniedoardo** [reply]
Decision from 10th Nov meeting: consider this update after 1,0 release as it would involve considerable work due to decisions regarding families, reviews of all functions, and new documentation of these families.

**Comment by bms63** [comment]
I added this into the Roadmap 2.0 discussion with link and will close - https://github.com/pharmaverse/admiral/discussions/2191#discussioncomment-8127461
